tools/toolchains/__init__.py

```diff
@@ -733,12 +733,27 @@ def add_regions(self):
             ))
             self._add_all_regions(regions, "MBED_RAM")
         try:
-            rom_start, rom_size = self.config.rom
-            Region = namedtuple("Region", "name start size")
-            self._add_defines_from_region(
-                Region("MBED_ROM", rom_start, rom_size),
-                suffixes=["_START", "_SIZE"]
-            )
+            # Add all available ROM regions to build profile
+            rom_available_regions = self.config.get_all_rom_regions(True)
+            for key, value in rom_available_regions.items():
+                rom_start, rom_size = value
+                Region = namedtuple("Region", "name start size")
```

Comment:
Would you mind moving this declaration out of the loop? (it'd probably be a good idea to move it out of the try/except block as well)